chain/client/src/client.rs

```diff
@@ -2010,8 +2020,22 @@ impl Client {
             } else {
                 // Transactions only need to be recorded if the node is a validator.
                 if me.is_some() {
-                    self.sharded_tx_pool.insert_transaction(shard_id, tx.clone());
-                    trace!(target: "client", shard_id, "Recorded a transaction.");
+                    match self.sharded_tx_pool.insert_transaction(shard_id, tx.clone()) {
+                        InsertTransactionResult::Success => {
+                            trace!(target: "client", shard_id, "Recorded a transaction.");
+                        }
+                        InsertTransactionResult::Duplicate => {
+                            trace!(target: "client", shard_id, "Duplicate transaction, not forwarding it.");
+                            return Ok(ProcessTxResponse::ValidTx);
+                        }
+                        InsertTransactionResult::NoSpaceLeft => {
+                            if is_forwarded {
+                                trace!(target: "client", shard_id, "No space left for transaction, dropping it.");
```

Comment:
I'd prefer wording as `Transaction pool is full, dropping a transaction`.

Reply:
I like the suggestion, changed.